x/amm/keeper/apply_join_pool_state_change.go

```diff
@@ -17,6 +24,29 @@ func (k Keeper) applyJoinPoolStateChange(ctx sdk.Context, pool types.Pool, joine
 
 	k.SetPool(ctx, pool)
 
+	rebalanceTreasuryAddr := sdk.MustAccAddressFromBech32(pool.GetRebalanceTreasury())
+
+	if weightBalanceBonus.IsPositive() {
+		// calculate treasury amounts to send as bonus
+		weightBalanceBonusCoins := PortionCoins(joinCoins, weightBalanceBonus)
+		for _, coin := range weightBalanceBonusCoins {
+			treasuryTokenAmount := k.bankKeeper.GetBalance(ctx, rebalanceTreasuryAddr, coin.Denom).Amount
```

Comment:
so a single recipient will get all the balance in treasury ? is this right ?

Reply:
@amityadav0 only if the calculated `weightBalanceBonusCoins` execeeds the rebalance treasury balance yes